Mikrotik cAP ac patchset

vibornoff opened the pull request.

alaraun committed:
ipq40xx: add MikroTik cAP ac support

This adds support for the MikroTik RouterBOARD RBcAPGi-5acD2nD
(cAP ac), a  indoor dual band, dual-radio 802.11ac wireless AP, two
10/100/1000 Mbps Ethernet ports.

See https://mikrotik.com/product/cap_ac for more info.

Specifications:
 - SoC: Qualcomm Atheros IPQ4018
 - RAM: 128 MB
 - Storage: 16 MB NOR
 - Wireless:
   · Built-in IPQ4018 (SoC) 802.11b/g/n 2x2:2, 2.5 dBi antennae
   · Built-in IPQ4018 (SoC) 802.11a/n/ac 2x2:2, 2.5 dBi antennae
 - Ethernet: Built-in IPQ4018 (SoC, QCA8075) , 2x 1000/100/10 port,
   PoE in and passive PoE out

Unsupported:
 - PoE out

Installation:
Boot the initramfs image via TFTP and then flash the sysupgrade
image using "sysupgrade -n"

Signed-off-by: Alar Aun <alar.aun@gmail.com>

vibornoff committed:
ipq40xx: add MikroTik cAP ac better LEDs support

Make ethernet port LEDs availabale to the linux leds subsystem. Both
port LEDs are exposed as `green:eth1` and `green:eth2` respectively
under `/sys/class/leds/`

Move panic indication to "user" LED. This is due to the fact that
`GPIO 0` is not solely dedicated to drive "power" LED but to act
as a key for both "power" and "poe out" LEDs. This behavior may
potentially confuse end-user once he sees either blinking blue LED
or (when PoE device plugged) both blue and red LEDs.

Remove "poeout" LED definition from DTS. This LED isn't directly
controllable: `GPIO 2` controls PoE power supply for a plugged PoE
device and `GPIO 0` controls both "power" and "poeout" LEDS:
+-------------------+-----------+------------+--------------+
| GPIO state        | power led | poeout led | power supply |
+-------------------+-----------+------------+--------------+
| 0: on, 2: off     | on        | off        | off          |
| 0: on, 2: on      | on        | on         | on           |
| 0: off, 2: off    | off       | off        | off          |
| 0: off, 2: on     | off       | off        | on           |
+-------------------+-----------+------------+--------------+

Declare controllable LEDs in `/etc/board.d/01_leds` to be visible
to LuCi out of the box.

Change key code for "mode" button to "KEY_LIGHTS_TOGGLE" for
easier "Dark mode" implementation.

vibornoff commented:
@alaraun Any move on this PR?

alaraun commented on target/linux/ipq40xx/base-files/etc/board.d/01_leds:
```diff
@@ -45,6 +45,14 @@ engenius,ens620ext)
+mikrotik,cap-ac)
+	ucidef_set_led_default "power" "POWER" "blue:power" "1"
+	ucidef_set_led_default "user" "USER" "green:user" "0"
+	ucidef_set_led_netdev "eth1" "ETH1" "green:eth1" "eth1"
```
It is confusing to have different eth names for the same eth devices in different places.

vibornoff replied:
Take a look into a diagram:
```
  CPU       switch0
+------+   +--------+
| eth0 |---| port 0 |
| eth1 |   | port 1 |
+------+   | port 2 |                Enclousure
   |       | port 3 |     PHY          label
   |       | port 4 |---ethphy3---< E2/PoE out
   +-------| port 5 |---ethphy4---< E1/PoE in
           +--------+
```

So LEDs are named according to its enclouse labels:
* "green:eth1" for "E1/PoE in" labeled port
* "green:eth2" for "E2/PoE out" labeled port

Attaching LED triggers to switch port events didn't work for some reason. So I attached to netdev events and it worked like a charm:
* "green:eth1" to "eth1" device
* "green:eth2" to "eth0" device

Any idea on how to get it working better?

alaraun commented:
I have built the image with the current changes and I was not able to get dynamic IP from lan interface after flashing.

vibornoff commented:
> I have built the image with the current changes and I was not able to get dynamic IP from lan interface after flashing.

Not sure but isn't it a default behavior of all fresh OpenWRT setups? If I remember it right I had to configure static IP from 192.168.1.0/24 on my laptop to get access to my old WDR4300 after flashing.

alaraun commented:
> > I have built the image with the current changes and I was not able to get dynamic IP from lan interface after flashing.
> 
> Not sure but isn't it a default behavior of all fresh OpenWRT setups? If I remember it right I had to configure static IP from 192.168.1.0/24 on my laptop to get access to my old WDR4300 after flashing.

eth2 does not respond even when setting computer IP to static 192.168.1.x. It mean cAP ac is not usable by default build.

vibornoff commented:
> eth2 does not respond even when setting computer IP to static 192.168.1.x. It mean cAP ac is not usable by default build.

Yep, my fault. Switch-related patch is not yet ready. I'll rearrange my commits shortly with only LEDs-related stuff.

vibornoff committed:
ipq40xx: add Mikrotik cAP ac switch support

- Disable EDMA driver vlan tagging
- Remove virtual `eth1` device
- Change `green:ethX` LED triggers to switch port events
- Put default switch config into `02_network`

vibornoff commented:
Switch patch seems ready. VLANs work too.
However there is one minor issue. E1/E2 don't blink on switch port tx/rx for some reason, only link status events trigger those leds to lit on and off.

vibornoff commented:
Hello! Hope you have ever chance to take a look to this PR. It is absolutely viable to be merged, especially switch-related patch.
BTW, my awesome WDR4300 didn't survive a power surge so I had to flash cAP-ac, and it just works.
